x/wasm/internal/keeper/querier.go

```diff
@@ -76,13 +88,15 @@ func queryContractState(ctx sdk.Context, bech string, req abci.RequestQuery, kee
 
 	var state []types.Model
 	for ; iter.Valid(); iter.Next() {
+		if !accept(iter.Key()) {
```

Comment:
Iterate over all keys just to get one? Super inefficient.
The db supports `Get`. You just have to expose this in the `Keeper`.
Feel free to add/refactor functions there to make the querier cleaner